src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/StaffRefDataServiceImpl.java

```diff
@@ -796,17 +783,13 @@ public CaseWorkerProfile processExistingCaseWorkers(
             if (isUserSuspended(UserProfileUpdatedData.builder().idamStatus(IDAM_STATUS_SUSPENDED).build(),
                     caseWorkerProfiles.getCaseWorkerId(), ORIGIN_EXUI)) {
                 caseWorkerProfiles.setSuspended(true);
-                filteredProfile = caseWorkerProfiles;
-                return filteredProfile;
             }
-        } else {
-            //when existing profile with delete flag is false then update user in CRD db and roles in SIDAM
-            filteredProfile = updateSidamRoles(caseWorkerProfiles, cwUiRequest);
-            return filteredProfile;
         }
-        //add user roles in user profile and filter out UP failed records
+        filteredProfile = updateSidamRoles(caseWorkerProfiles,cwUiRequest);
```

Comment:
calling update sidam roles method all the time will impact performance. ex: if user is only suspending then we no need to update his roles again

Reply:
agreed with @sahitya-desireddy comments . resolving this comment